server/server.go

```diff
@@ -0,0 +1,88 @@
+package server
+
+import (
+	"context"
+	"net/http"
+	"sync"
+
+	"github.com/giantswarm/microerror"
+	microserver "github.com/giantswarm/microkit/server"
+	"github.com/giantswarm/micrologger"
+	kithttp "github.com/go-kit/kit/transport/http"
+
+	"github.com/giantswarm/app-operator/service"
+)
+
+// Config represents the configuration used to construct server object.
+type Config struct {
+	// Dependencies
+	Service *service.Service
+
+	//Settings
+	MicroServerConfig microserver.Config
+}
+
+// New creates a new server object with given configuration.
+func New(config Config) (microserver.Server, error) {
+	if config.Service == nil {
+		return nil, microerror.Maskf(invalidConfigError, "config.Service must not be empty")
```

Comment:
```suggestion
		return nil, microerror.Maskf(invalidConfigError, "%T.Service must not be empty", config)
```

Reply:
FYI you can see how this is usually done in https://github.com/giantswarm/fmt/blob/master/go/struct_initialization.md.